ui/redux/rootReducer.js

```diff
@@ -370,6 +371,7 @@ export const updateIgvReadsVisibility = updates => ({ type: UPDATE_IGV_VISIBILIT
 const rootReducer = combineReducers(Object.assign({
   projectCategoriesByGuid: createObjectsByIdReducer(RECEIVE_DATA, 'projectCategoriesByGuid'),
   projectsByGuid: createObjectsByIdReducer(RECEIVE_DATA, 'projectsByGuid'),
+  googleAuthLoading: loadingReducer(REQUEST_GOOGLE_AUTH_RESULT, RECEIVE_DATA),
```

Comment:
You should use a different action for when a response is received, `RECEIVE_DATA` triggers a ton of other events to dispatch unnecessarily. Why not just `RECEIVE_GOOGLE_AUTH_RESULT`?

Reply:
`RECEIVE_GOOGLE_AUTH_RESULT` is a good alternative for `RECEIVE_DATA`. But I probably move all those to Django Views.